### PR TITLE
datanode viewer WiP no chart (#251)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
a demo that uses all components
```
from taipy.gui_core.GuiCoreLib import _GuiCore

from taipy.gui import Gui, State
from taipy import Config, Core
import taipy as tp

def on_creation(state: State, id: str, action: str, payload: dict):
    print(f"on_creation(state: State, id: str, action: str, payload: dict) => {payload.get('label')}")

# Normal function used by Taipy
def d(nb):
    return nb * 2

i_cfg = Config.configure_data_node("input", default_data=21)
o_cfg = Config.configure_data_node("output")

task_cfg = Config.configure_task("double",d,i_cfg,o_cfg)

scenario_cfg = Config.configure_scenario("my_scenario", [task_cfg])

selected_scenario = None
none_scenario = None
selected_data_node = None
selected_jobs = None

if __name__ == '__main__':
    # Run of the Core
    Core().run()

    # Creation of the scenario and execution
    selected_scenario = tp.create_scenario(scenario_cfg)
    tp.submit(selected_scenario)

    md = """
<|{selected_scenario}|scenario_selector|>
<|{selected_scenario}|scenario|>
<|{selected_scenario}|scenario_dag|>
"""

    page = """
## Demo Core Gui
<|toggle|theme|>

<|layout|columns=1 1|

<|
### Scenario Selector
<|{selected_scenario}|scenario_selector|on_creation=on_creation|>
|>

<|
### DataNode Selector
<|{selected_data_node}|data_node_selector|>
|>

<|
### Visualizer
<|{selected_scenario}|scenario|>
|>

<|
### Selected DataNode
<|{'None' if selected_data_node is None else selected_data_node.get_label()}|>
<|{selected_data_node}|data_node|scenario={selected_scenario}|>
<|{none_scenario}|data_node|>
|>


### DAG
<|{selected_scenario}|scenario_dag|>

|>

### Job Selector
<|{selected_jobs}|job_selector|>


"""

    Gui(page).run(debug=True)

```

### CODE REVIEW

File: src/taipy/gui_core/viselements.json [hunk 1]
```diff
@@ -61,6 +61,12 @@
...
                         "default_value": "False",
                         "doc": "If True, a pin is shown on each item of the selector and allows to restrict the numnber of displayed items."
+                    },
+                    {
+                      "name": "on_creation",
+                      "type": "Callback",
+                      "doc": "The name of the function that is triggered when a scenario will be or has been created.<br/><br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the identifier of the element.</li>\n<li>action (str): the name of the action that provoked the change.</li>\n<li>payload (dict): the details on this callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>config: the selected configuration.</li>\n<li>date: the selected date.</li>\n<li>name: the user entered name.</li>\n<li>properties: a list of key/value dictionnaries.</li>\n</ul>\n</li>\n<li>return: the callback can return a scenario, an error message (the scenario will not be created) or None (the scenario will be created with the user parameters).</li>\n</ul>",
```
Comment (FabienLelaquais) [inline src/taipy/gui_core/viselements.json]: "when a scenario will be" -> "when a scenario is about to be"
  remove the "or has been" part
"identifier of the element" -> "identifier of the scenario selector"
"action that provoked the change" -> "action that provoked the invocation of this callback"
   (is this the name of this callback function?)
"the selected configuration" -> "the name of the selected scenario configuration"
"the selected date" -> "the creation date for the new scenario" (is it a Date?)
"a list of key/value dictionnaries" ?? a dictionary?
"an error message" -> "a string containing an error message"
"the scenario will be created" -> "then a new scenario is created"

File: gui/src/DataNodeViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+    tabularData?: TableValueType;
+    tabularColumns?: string;
+    onDataValue?: string;
+    onTabularDataEdit?: string;
+}
+
+const getDataValue = (value?: RowValue, dType?: string | null) => (dType == "date" ? new Date(value as string) : value);
+
+const getValidDataNode = (datanode: DataNodeFull | DataNodeFull[]) =>
+    datanode.length == DataNodeFullLength && typeof datanode[DataNodeFullProps.id] === "string"
```
Comment (FabienLelaquais) [inline gui/src/DataNodeViewer.tsx:140]: Can DataNodeFullLength be 0??

**Review by FabienLelaquais — DISMISSED** [review]
Too much for me at this point...

File: gui/src/DataNodeViewer.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply gui/src/DataNodeViewer.tsx:140]: nope

File: src/taipy/gui_core/viselements.json (hunk 1 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui_core/viselements.json]: added data_node deoc

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**